src/charm.py

```diff
@@ -405,14 +405,24 @@ def _configure_cni(self):
 
     def _get_cni_config(self):
         """Get the CNI config options."""
+        binding = self.model.get_binding("cni")
+
+        ipv4_cidr = next(
+            (
+                iface.subnet
+                for iface in binding.network.interfaces
+                if not iface.name.startswith("fan-")
+            ),
+            None,
+        )
```

Comment:
is it okay if this is `None` since your list of interfaces could be empty?  If it will never be `None`

you could consider this simplification

```suggestion
        ipv4_cidr, *_ = (
            iface.subnet
            for iface in binding.network.interfaces
            if not iface.name.startswith("fan-")
        )
```